xen/include/public/io/sndif.h

```diff
@@ -56,72 +56,155 @@
  *                            Backend XenBus Nodes
  *****************************************************************************
  *
- *------------------------- Backend Device parameters -------------------------
+ *-------------------------------- Addressing ---------------------------------
  *
- * devid
- *      Values:         <uint32_t>
+ * Indices used to address frontends, driver instances, cards,
+ * devices and streams.
  *
- *      Index of the soundcard which will be created in the frontend. This
- *      index is zero based.
+ * frontend-id
+ *      Values:         <uint>
  *
- * devcnt
- *      Values:         <uint32_t>
+ *      Domain ID of the sound frontend.
  *
- *      PCM instances count that are created by the soundcard in the frontend.
+ * drv_idx
+ *      Values:         <uint>
  *
- *----------------------------- Streams settings ------------------------------
+ *      Zero based index of the virtualized sound driver instance in this domain.
+ *      Multiple PV drivers are allowed in the domain at the same time.
  *
- * Every virtualized device has own set of the sound streams. Each stream
- * parameter is with index "%u" and defined as 'stream%u_???'. Stream index is
- * zero based and should be continuous in range from 0 to 'streams_cnt' - 1.
+ * card_idx
+ *      Values:         <uint>
  *
- * stream%u_channels
- *      Values:         <uint32_t>
+ *      Zero based index of the card within the driver.
  *
- *      The maximum amount of channels that can be supported by this stream.
- *      Should be from 1 to XENSND_MAX_CHANNELS_PER_STREAM.
+ * dev_id
+ *      Values:         <uint>
  *
- * stream%u_type
- *      Values:         "p", "c", "b"
+ *      Unique (within given card instance) device ID.
+ *      Doesn't have to be zero based and/or be contiguous.
  *
- *      Stream type: "p" - playback stream, "c" - capture stream,
- *      "b" - both playback and capture stream.
+ * stream_idx
+ *      Values:         <uint>
  *
- * stream%u_bedev_p
- *      Values:         string
+ *      Zero based index of the stream of the device.
  *
- *      Name of the playback sound device which is mapped to this stream
- *      by the backend. Present if stream%u_type is "p" or "b"
+ * Example for the frontend running in domain 5, instance of the driver
+ * in the front is 0 (single or first PV driver), sound card 0, device id 2,
+ * first stream (0):
+ * /local/domain/<frontend-id>/device/vsnd/<drv_idx>/card/<card_idx>/
```

Comment:
I would say yes, as XenStore is targeted to be human readable and these strings help one understand what is this all about.
At the same time I do agree that this is memory waste